packages/slice-machine/src/legacy/lib/builders/CustomTypeBuilder/index.tsx

```diff
@@ -42,10 +43,17 @@ export const CustomTypeBuilder = () => {
     customTypeSM.tabs.find((tab) => tab.key === tabValue)?.sliceZone?.value
       .length === 0;
 
+  // block removing tab with UID field
```

Comment:
💡 #idea

I was lost to understand the comment, was reading block as a block of text or something like that 😅
Maybe we could move that line directly above where this is done, here you're not blocking the tab with UID field, this code is only to get the `tabWithUID`.

Also improving naming to ensure nobody face the same problem as me haha.

```
                   <DropdownMenuItem
                        color="tomato"
                        // Blocking the ability to remove a tab if it's the last one or it contains an UID.
                        disabled={
                          customTypeSM.tabs.length <= 1 ||
                          tabWithUid?.key === tab.key
                        }
```